### PR TITLE
Fixes resourceServiceBuilder to work on Windows

### LINKED ISSUE #5240: ResourceServiceBuilder using OS-dependent file paths instead of resource paths
https://github.com/http4s/http4s/blob/56c50a9c6ebc4969c707a19595820983b06ba6bf/server/src/main/scala/org/http4s/server/staticcontent/ResourceService.scala#L101

The above line invokes [Path#toString](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Path.html#toString()), which emits a string using the default OS path separator. However, that string path is then used to load the resource file with [ClassLoader#getResource](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/ClassLoader.html#getResource(java.lang.String)), which expects all paths to use a '/' separator.

### PR DESCRIPTION
**Pull request by mabasic**
Fixes #5240 

I have tested the fix on Windows and Linux, and it works.

### CODE REVIEW

File: server/shared/src/main/scala/org/http4s/server/staticcontent/ResourceService.scala [hunk 1]
```diff
@@ -103,7 +103,7 @@ class ResourceServiceBuilder[F[_]] private (
               }
               .flatMap { path =>
                 StaticFile.fromResource(
-                  path.toString,
+                  path.toString.replaceAll("\\\\", "/"),
```
Comment (satorg) [inline server/shared/src/main/scala/org/http4s/server/staticcontent/ResourceService.scala:106]: I wonder if it makes sense to be more OS-specific here.
E.g. on JVM there's a `java.io.File.separator` constant which can be used here for the sake of the replacement.

Comment (mabasic) [reply server/shared/src/main/scala/org/http4s/server/staticcontent/ResourceService.scala:106]: As I understand it is that we need to provide Linux separator regardless if the code runs on Windows or Linux. Do you have a suggestion on what to change?

Comment (armanbilge) [reply server/shared/src/main/scala/org/http4s/server/staticcontent/ResourceService.scala:106]: Right, Sergey's proposal is not to use `java.io.File.separator` for the _replacement_, but for the pattern to replace i.e. the `"\\\\"`.

Comment (satorg) [reply server/shared/src/main/scala/org/http4s/server/staticcontent/ResourceService.scala:106]: Yes, correct. Sorry if I made it unclear. Basically, if we assume that path segment separator is always a single char regardless of the target OS, then we can make use of
```java
public String replace(char oldChar, char newChar)
```
from `String`:
```suggestion
                  path.toString.replace(File.separatorChar, '/'),
```
I guess it should be faster that `replaceAll` (which takes a regex for the first parameter).
Also, `replace` does nothing (no-op) if `oldChar` and `newChar` are the same.

Comment (armanbilge) [reply server/shared/src/main/scala/org/http4s/server/staticcontent/ResourceService.scala:106]: Oh, great that it's `separatorChar`! That's very nice :)

Comment (mabasic) [reply server/shared/src/main/scala/org/http4s/server/staticcontent/ResourceService.scala:106]: Very nice. I like your solution better. I will try this and test on Windows and Linux if it works just to be safe.

Comment (mabasic) [reply server/shared/src/main/scala/org/http4s/server/staticcontent/ResourceService.scala]: I have tested it and it works both on Windows and Linux.

**Review by armanbilge — APPROVED** [review]

**Review by satorg — APPROVED** [review]

**Comment by mabasic** [comment]
@satorg Any idea, when the new release will be :) ?

**Comment by armanbilge** [comment]
@mabasic you can use the snapshot version: `0.23.16-266-5c8acee-SNAPSHOT`.

Make sure to add:
```scala
ThisBuild / resolvers += Resolver.sonatypeOssRepos("snapshots")
```

**Pull request merged**